bundles/org.openhab.binding.bluetooth.bluegiga/src/main/java/org/openhab/binding/bluetooth/bluegiga/internal/BlueGigaDeviceResponse.java

```diff
@@ -0,0 +1,44 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.bluetooth.bluegiga.internal;
+
+/**
+ * Abstract base class for all device responses.
+ *
+ * @author Pauli Anttila - Initial contribution
+ *
+ */
+public abstract class BlueGigaDeviceResponse extends BlueGigaResponse {
+
+    /**
+     * Connection handle.
+     * <p>
+     * BlueGiga API type is <i>uint8</i> - Java type is {@link int}
+     */
+    protected Integer connection = null;
```

Comment:
I think this should be a primitive `int` instead since all the refactored classes expect the primitive as well. You might run into NPEs since most of the code that calls `getConnection` hasn't taken into account that it can now return nulls.

Reply:
Indeed, this is left over from some test. I already changed the BlueGigaDeviceCommand back to int, but forgotten the change this.